source/common/common/logger.cc

```diff
@@ -126,15 +128,33 @@ Context::~Context() {
   }
 }
 
-void Context::activate(LoggerMode mode) {
+void Context::activate() {
   Registry::getSink()->setLock(lock_);
   Registry::getSink()->setShouldEscape(should_escape_);
   Registry::setLogLevel(log_level_);
   Registry::setLogFormat(log_format_);
 
-  if (mode == LoggerMode::Fancy) {
-    fancy_default_level_ = log_level_;
-    fancy_log_format_ = log_format_;
+  // sets level and format for Fancy Logger
+  fancy_default_level_ = log_level_;
+  fancy_log_format_ = log_format_;
+  logger_mode_ = log_mode_;
+  if (logger_mode_ == 1) {
+    // loggers with default level before are set to log_level_ as new default
+    getFancyContext().setDefaultFancyLevelFormat(log_level_, log_format_);
+  }
+}
+
+LoggerMode Context::getLoggerMode() { return static_cast<LoggerMode>(logger_mode_); }
```

Comment:
Why do you need both a static logger_mode_ and a member log_mode_?

I'd imagine you have just a getLoggerMode() that returns current_context->log_mode_, which should be an enum.